src/components/main-view/main-view.jsx

```diff
@@ -23,6 +24,7 @@ export const MainView = () => {
   const [search, setSearch] = useState("");
   const [selectedGenre, setSelectedGenre] = useState("");
   const [selectedMovie, setSelectedMovie] = useState(null);
+  const [loading, setLoading] = useState(true); // adding state for loading
```

Comment:
By default `loading` should be false, and it should only be set to true just before we make the API call. So you call `setLoading(true);` on line 35 on this file.